packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1393,8 +1391,13 @@ export async function renderToHTMLOrFlight(
                 initialCanonicalUrl={pathname}
                 initialTree={initialTree}
                 initialHead={
-                  // Adding requestId as react key to make metadata remount for each render
-                  <MetadataTree key={requestId} />
+                  <>
+                    {res.statusCode > 400 && (
```

Comment:
As we can render root not-found with a `__PAGE__` segment now, check if we need robots meta tag here. Ideally in the future we could move more control into metadata rendering